### PR TITLE
[#519] correct regression with server_verify_mode of "none"

### PR DESCRIPTION
**Pull request by d-w-moore**

### CODE REVIEW

File: irods/connection.py
Comment (korydraughn) [inline irods/connection.py:187]: How do I interpret `None is CAfile is CApath`?

File: irods/connection.py
Comment (korydraughn) [inline irods/connection.py:191]: Should we check for `hostname` rather than `host`?

Is it safe to assume `verify_server`'s full value for the true case is supposed to be `hostname`?

File: irods/test/login_auth_test.py
Comment (korydraughn) [inline irods/test/login_auth_test.py:247]: Let's be explicit about which values we're checking for to better future-proof the software.

If a third PAM value appears later on for iRODS, could this result in incorrect behavior?

File: irods/test/login_auth_test.py
Comment (korydraughn) [inline irods/test/login_auth_test.py:331]: Whitespace

**Review by korydraughn — COMMENTED** [review]
Seems good.

File: irods/connection.py
Comment (alanking) [reply irods/connection.py:187]: I believe the following are all equivalent:
 - `None is CAfile is CApath`
 - `None is CAfile and None is CApath`
 - `CAfile is None and CApath is None`
 - `None == CAfile == CApath`
 - `None == CAfile and None == CApath`

And you can imagine the remaining combinations. :)

File: irods/connection.py
Comment (alanking) [reply irods/connection.py:191]: Perhaps older iRODS versions used a different scheme (e.g. `hostName`, `host`, etc.)?

File: irods/test/login_auth_test.py
Comment (alanking) [inline irods/test/login_auth_test.py:205]: Should this be >=?
```suggestion
        if cls.admin.server_version >= (4,3):
```

File: irods/test/login_auth_test.py
Comment (alanking) [reply irods/test/login_auth_test.py:247]: I think this is to accommodate the fact that `pam` and `pam_password` are interchangeable right now in PRC for 4.3+ servers. Perhaps a comment to that effect can help future readers.

Also, tell me if I'm wrong here :p

File: irods/test/login_auth_test.py
Comment (alanking) [inline irods/test/login_auth_test.py:515]: Is it still the case that this directory need not exist?

File: irods/test/login_auth_test.py
Comment (korydraughn) [reply irods/test/login_auth_test.py:247]: Right. I was thinking about how the sub-string check doesn't guard against bogus values like `pamattack`. If a bogus value makes it here, it could be possible to break something or put the PRC in a weird state.

File: irods/connection.py
Comment (d-w-moore) [reply irods/connection.py:187]: Right.  Python is one of those weird languages where often `(a OP b OP c) `is not equivalent to either
```
(a OP b) OP c
```
or
```
a OP (b OP c)
```
For example, `a < b < c` means `(a < b) and (b < c)` ; this also ggoes for == , !=, etc

File: irods/test/login_auth_test.py
Comment (d-w-moore) [reply irods/test/login_auth_test.py:205]: I think they're equivalent because server_version will never report `(4,3)`.

File: irods/connection.py
Comment (d-w-moore) [reply irods/connection.py:191]: verify_server == 'host' is probably correct.

File: irods/connection.py
Comment (korydraughn) [reply irods/connection.py:187]: Thanks guys. That line had me cross-eyed for a second.

File: irods/connection.py
Comment (d-w-moore) [reply irods/connection.py:191]: > `verify_server == 'host'` is probably correct.

oops hostname.

Comment (d-w-moore) [reply irods/connection.py]: Corrected.

File: irods/test/login_auth_test.py
Comment (d-w-moore) [reply irods/test/login_auth_test.py]: This is corrected now.

File: irods/test/login_auth_test.py
Comment (d-w-moore) [reply irods/test/login_auth_test.py]: corrected

File: irods/test/login_auth_test.py
Comment (d-w-moore) [reply irods/test/login_auth_test.py]: Altering this to be a temporary directory, which it should have been to begin with.

File: irods/test/login_auth_test.py
Comment (d-w-moore) [reply irods/test/login_auth_test.py]: It won't hurt and will be arguably clearer I suppose. changing ....

Comment (d-w-moore) [reply irods/test/login_auth_test.py]: It won't hurt and will be arguably clearer I suppose. changing ....

**Review by korydraughn — COMMENTED** [review]
Seems good. Has this been tested?

**Comment by d-w-moore** [comment]
> Seems good. Has this been tested?

These changes were tested pre-review but not yet post-review.  Likely I'll wait until after review a few more PR's and testing the changes all together, then we'll merge them all at once.  That is just the more efficient way to do it at the moment, given the changes to the testing process that are in store for PRC v2.1.0

**Comment by korydraughn** [comment]
That sounds more error prone since you're handling multiple PRs, but if you feel that's the most efficient way to handle it on your end, then proceed.

**Comment by d-w-moore** [comment]
> That sounds more error prone since you're handling multiple PRs, but if you feel that's the most efficient way to handle it on your end, then proceed.

I felt it was the most efficient, if not overall best, way.

**Comment by d-w-moore** [comment]
Changing commit message to exclude #281 in head-line per https://github.com/irods/python-irodsclient/pull/508#issuecomment-1997374473

**Comment by d-w-moore** [comment]
Tested.  To resolve "critical" codacy comment (re unexpected parameter `ssl_dir`) this PR should follow #527 in the merge order.

**Comment by korydraughn** [comment]
Squash to taste if ready, no pounds.

We'll hold on merging this one until #527 is merged.

**Review by alanking — APPROVED** [review]

**Pull request merged**
